### PR TITLE
Feat/aapulskis

### PR DESCRIPTION
**Pull request by ArvApu**

### CODE REVIEW

File: Web/Controllers/Investments/RealEstateController.cs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        [HttpGet]
+        [Route("real-estates")]
+        public IActionResult OpenRealEstateList()
+        {
+            return View("RealEstateList", new RealEstateListViewModel() {RealEstates = SelectUserRealEstate()});
+        }
+        
+        [Route("real-estates/edit/{id:int}")]
+        [HttpGet]
+        public IActionResult OpenEditForm(int id)
```
Comment (EdotJ) [inline Web/Controllers/Investments/RealEstateController.cs:27]: Bad name, look at the diagram, should be EditRealEstate

File: Web/Controllers/Investments/RealEstateController.cs [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        {
+            var realEstate = repository.RealEstate.FirstOrDefault(x => x.Id == id);
```
Comment (EdotJ) [inline Web/Controllers/Investments/RealEstateController.cs:29]: Look at the diagram, should be separate method.

File: Web/Controllers/Investments/RealEstateController.cs [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+                Address = realEstate.Address,
+                IsRented = realEstate.IsRented,
+                SquareSpace = realEstate.SquareSpace,
+                RoomNumber = realEstate.RoomNumber,
+                Valuation = realEstate.Valuation,
+            });
+        }
+        
+        [HttpPost]
+        public IActionResult Edit(RealEstateViewModel viewModel)
```
Comment (EdotJ) [inline Web/Controllers/Investments/RealEstateController.cs:49]: Should be UpdateRealEstate

Comment (EdotJ) [reply Web/Controllers/Investments/RealEstateController.cs:49]: Looking at the diagrams, the naming is off. This or the method in the entity should be changed and the code updated.

File: Web/Controllers/Investments/RealEstateController.cs [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        
+            UpdateRealEstate(realEstate, viewModel);
+        
+            var realEstates = SelectUserRealEstate();
+        
+            return View("RealEstateList", new RealEstateListViewModel() {RealEstates = realEstates});
+        }
+        
+        [HttpGet]
+        public IActionResult OpenCreationForm()
```
Comment (EdotJ) [inline Web/Controllers/Investments/RealEstateController.cs:79]: OpenRealEstateForm

File: Web/Controllers/Investments/RealEstateController.cs [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+        {
+            return View("RealEstateForm", new RealEstateViewModel());
+        }
+        
+        [HttpPost]
+        public IActionResult Create(RealEstateViewModel viewModel)
```
Comment (EdotJ) [inline Web/Controllers/Investments/RealEstateController.cs:85]: CreateRealEstate. Should also be an operation in the diagram. Let me know if you'll fix that or if I should do it.

File: Web/Controllers/Investments/RealEstateController.cs [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
+            if (viewModel.Valuation <= 0)
+            {
+                return "Vertė turi būti daugiau už 0.";
+            }
+            
+            return string.Empty;
+        }
+        
+        private IList<DataAccess.Models.RealEstate> SelectUserRealEstate()
```
Comment (EdotJ) [inline Web/Controllers/Investments/RealEstateController.cs:202]: SelectUsersRealEstate

File: Web/Controllers/Investments/RealEstateController.cs [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+        
+            TempData["Success"] = "NT sėkmingai pridėtas!";
+        
+            var realEstates = SelectUserRealEstate();
+        
+            return View("RealEstateList", new RealEstateListViewModel() {RealEstates = realEstates});
+        }
+        
+        [HttpGet]
+        public IActionResult Delete(int id)
```
Comment (EdotJ) [inline Web/Controllers/Investments/RealEstateController.cs:106]: RemoveRealEstate

File: Web/Controllers/Investments/RealEstateController.cs [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+        {
+            return PartialView("DeleteConfirmForm",
+                new DeletionViewModel {LtName = "nekilnojamas turtas", Id = id, Controller = "RealEstate", Method = "DeleteRealEstate"});
+        }
+        
+        [HttpDelete]
+        public IActionResult DeleteRealEstate(int id)
```
Comment (EdotJ) [inline Web/Controllers/Investments/RealEstateController.cs:113]: DeleteSelected

File: Web/Controllers/Investments/RealEstateController.cs [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+        {
+            var realEstate = repository.RealEstate.FirstOrDefault(x => x.Id == id);
+        
+            repository.Remove(realEstate);
```
Comment (EdotJ) [inline Web/Controllers/Investments/RealEstateController.cs:117]: Repository changes should be executed in a different method called DeleteSelected. This conflicts with the comment above, please change the diagram accordingly

File: Web/Views/RealEstate/RealEstateForm.cshtml [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+    var isAdd = Model.Id == 0;
+    var title = isAdd ? "Pridėti nekilnojamą turtą" : "Redaguoti nekilnojamą turtą";
+
+    ViewData["Title"] = title;
+}
+
+<h1>@(title)</h1>
+
+@using (Html.BeginForm(isAdd ? "Create" : "Edit", "RealEstate", FormMethod.Post))
```
Comment (EdotJ) [inline Web/Views/RealEstate/RealEstateForm.cshtml:13]: Update this after you're done with Controller changes.

**Review by EdotJ — CHANGES_REQUESTED** [review]
Please review the diagrams and make changes to them / the code accordingly. Thank you! Everything else looks good to my untrained eye, maybe Rokas could offer some deeper comments.

File: Web/Controllers/Investments/RealEstateController.cs (hunk 1 above)
Comment (ArvApu) [reply Web/Controllers/Investments/RealEstateController.cs:27]: šita pakeičiau diagramoje

File: Web/Controllers/Investments/RealEstateController.cs (hunk 2 above)
Comment (ArvApu) [reply Web/Controllers/Investments/RealEstateController.cs:29]: Ta pati klaida palikat kategorijose

File: Web/Controllers/Investments/RealEstateController.cs (hunk 3 above)
Comment (ArvApu) [reply Web/Controllers/Investments/RealEstateController.cs:49]: Pakeiciau diagramoje

File: Web/Controllers/Investments/RealEstateController.cs (hunk 4 above)
Comment (ArvApu) [reply Web/Controllers/Investments/RealEstateController.cs:79]: Pakeiciau diagramoje

File: Web/Controllers/Investments/RealEstateController.cs (hunk 6 above)
Comment (ArvApu) [reply Web/Controllers/Investments/RealEstateController.cs:202]: pakeiciau diagramoje

File: Web/Controllers/Investments/RealEstateController.cs (hunk 7 above)
Comment (ArvApu) [reply Web/Controllers/Investments/RealEstateController.cs:106]: Fixed

**Comment by ArvApu** [comment]
https://github.com/t-eliks/finsys/pull/6 SOLUTION

**Pull request closed without merging**